backend/pong/tmp_jwt/jwt.py

```diff
@@ -34,5 +36,66 @@ def __init__(self, payload: Optional[dict] = None) -> None:
         """
         # todo: アルゴリズムも選べるようにする？
         self.header: dict = {"typ": "JWT", "alg": "HS256"}
-        self.payload: dict = payload or {}
-        self.jws: jws.JWS = jws.JWS()
+        # todo:　エンコーディング方式をbase64urlに指定する
+        self.jws_handler: jws.JWS = jws.JWS()
+        self.base64_url_handler: base64_url.Base64Url = base64_url.Base64Url()
+
+    def encode(self, payload: dict) -> str:
+        """
+        引数で渡されたペイロードを検証し、署名済みのJWTを生成します。
+
+        Args:
+            todo: ペイロードの詳細な構造を記述
+            payload (dict)
+                - 必須フィールド: ...
+                - 任意フィールド: ...
+        Returns:
+            str: 署名済みのJWT
+
+        Raises:
+            ValueError:
+            - ペイロードが空である場合
+            - ペイロード内の必須フィールドが不足している場合
+            - ...
+        """
+        # todo: payloadの検証
+        encoded_header: str = self.base64_url_handler.encode_dict(self.header)
+        encoded_payload: str = self.base64_url_handler.encode_dict(payload)
+        encoded_signature: str = self.jws_handler.sign(
+            encoded_header, encoded_payload
+        )
+        return f"{encoded_header}.{encoded_payload}.{encoded_signature}"
+
+    def decode(self, jwt: str) -> dict:
+        """
+        引数で渡されたJWTを検証し、有効期限が保証されたペイロードをデコードして取得します。
+        Args:
+            jwt (str): 検証対象のJWT。
+
+        Returns:
+            dict: 有効期限が保証されたデコード済みのペイロード。
+
+        Raises:
+            ValueError:
+            - .の数が2個未満のJWTの形式の場合
+            - JWTの検証が失敗した場合
+            - JWTが有効期限切れの場合。
+        """
+        try:
+            header, payload, signature = jwt.split(".")
+        except ValueError:
+            logger.error(jwt)
+            raise ValueError(
+                "Invalid JWT format: must contain exactly two dots"
+            )
+        if self.jws_handler.verify(header, payload, signature):
+            decoded_payload: dict = self.base64_url_handler.decode_dict(
+                payload
+            )
+            # todo: payloadの有効期限が切れていないかどうか
+            return decoded_payload
+        else:
+            logger.error(jwt)
+            raise ValueError(
+                "JWT Signature verification failed: Invalid or tampered signature"
+            )
```

Comment:
[imo]
前のレビューで気付かず、全然いつでも良いのですが、早期リターンの方がネストが減らせて良さそうだと思いました
todo にこれから実装も増えると書いてあるので、if 文の中が肥大化するより良いかもです
```suggestion
        if not self.jws_handler.verify(header, payload, signature):
            logger.error(jwt)
            raise ValueError(
                "JWT Signature verification failed: Invalid or tampered signature"
            )
        decoded_payload: dict = self.base64_url_handler.decode_dict(
            payload
        )
        # todo: payloadの有効期限が切れていないかどうか
        return decoded_payload
```

Reply:
ありがとうございます！適用します！